mirgecom/viscous.py

```diff
@@ -275,3 +284,37 @@ def get_viscous_cfl(discr, eos, dt, cv):
         The CFL at each node.
     """
     return dt / get_viscous_timestep(discr, eos=eos, cv=cv)
+
+
+def get_local_max_species_diffusivity(actx, discr, d_alpha):
+    """Return the maximum species diffusivity at every point.
+
+    Parameters
+    ----------
+    actx: :class:`arraycontext.ArrayContext`
+        Array context to use
+    discr: :class:`grudge.eager.EagerDGDiscretization`
+        the discretization to use
+    d_alpha: np.ndarray
+        Species diffusivities
+    """
+    return_dof = []
+    for i in range(len(d_alpha[0])):
```

Comment:
Based on a discussion in our slack: should we explicitly check for `d_alpha` being `np.empty`?

Reply:
(4f95721)

Reply:
It's not np.empty that's the problem, it's of size 0. See my last change

Reply:
`d_alpha is not np.empty` isn't going to do what you want (it will check whether `d_alpha` is the function `np.empty`). `d_alpha.size > 0` or `len(d_alpha) > 0` should work.

Reply:
Perfect, thanks @majosm.  Do you know if there is a performance difference?   Does len(d_alpha) have constant time? I guess I suspect not if it is allowed to be non-contiguous.

Reply:
For 1D arrays I would think they'd be the same (`len` just calls `ndarray.__len__()`, which I assume just returns `shape[0]`). For higher dimensional arrays `size` returns a different result (it's the product of `shape`).